distributed/scheduler.py

```diff
@@ -6201,31 +6201,27 @@ def adaptive_target(self, comm=None, target_duration=None):
             self.total_occupancy / target_duration
         )  # TODO: threads per worker
 
-        # Avoid a few long tasks from asking for many cores
-        ws: WorkerState
-        tasks_processing = 0
-        for ws in self.workers.values():
-            tasks_processing += len(ws._processing)
-
-            if tasks_processing > cpu:
-                break
-        else:
-            cpu = min(tasks_processing, cpu)
-
         if self.unrunnable and not self.workers:
             cpu = max(1, cpu)
 
-        # Memory
-        limit_bytes = {addr: ws._memory_limit for addr, ws in self.workers.items()}
-        worker_bytes = [ws._nbytes for ws in self.workers.values()]
-        limit = sum(limit_bytes.values())
-        total = sum(worker_bytes)
-        if total > 0.6 * limit:
+        # add more workers if more than 60% of memory is used
+        limit = sum(ws.memory_limit for ws in self.workers.values())
+        used = sum(ws.nbytes for ws in self.workers.values())
+        memory = 0
+        if used > 0.6 * limit:
```

Comment:
it might be good to change this to `used > 0.6 * limit and limit > 0`. With no memory limit set, it tries to always increase the number of workers (that is why I initially thought it was related to memory).